### PR TITLE
Increase rest api integration coverage to match api offerings

### LINKED ISSUE #412: REST API integration tests for all parameters
**Problem**
We keep having regressions in the REST API because we don't have good test coverage. We have an integration test framework but it is not filled out with specs for all URLs and combinations of parameters.

In addition - the way we setup integration test data, execute integration tests, and compare results, is not intuitive/clear.

**Solution**
- Refactor the integration_test.js such that each test:
  - ~clears out the database of data before each test (see cleanup.sql), using jest's `beforeEach`~
  - ~specifies a test data file to use to setup database data at the beginning of the test (see integration_test_data.json and related logic in integration.test.js). The data setup files can be moved into a new `data` folder.~
  - ~`specs` can specify which data file they will use in the spec's json (`data_setup_file`: _path_)~
  - ~run the test against the target code (reqToSql or calling the REST API)~
  - ~compare the results to a result file and output/assert differences~
- Add spec files to `hedera-mirror-rest/__tests__/specs` for every URL and parameters we support. Also add negative test cases.

**Alternatives**

**Additional Context**
integration.test.js contains tests for both the end-to-end REST API-through-DB and reqToSql-through-DB. Don't split the file up into multiple files without ensuring that the tests do not run in parallel, as there is only 1 integration test DB.

### PR DESCRIPTION
**Pull request by Nana-EC**
**Detailed description**:
As noted in #412 "We keep having regressions in the REST API because we don't have good test coverage. We have an integration test framework but it is not filled out with specs for all URLs and combinations of parameters."

This PR increases the integration test coverage by providing parity coverage with The REST API endpoints and parameter options

- Updated integrationDomainOps.js to insert value for public key
- Added accounts-02-specific-id.spec.json to verify api/v1/accounts?account.id=x.y.z
- Added accounts-03-accounts-range.spec.json to verify account rage /api/v1/accounts?account.id=gte:x1.y1.y1&account.id=lt:x2.y2.z2
- Added accounts-04-balance-range.spec.json to verify /api/v1/accounts?account.balance=gt:x  and desc order param
- Added accounts-05-publickey.spec.json to verify /api/v1/accounts?account.publickey=x
- Added balances-02-specific-id.spec.json to verify /api/v1/balances?account.id=x.y.z
- Added balances-03-balance-range.spec.json to verify /api/v1/balances?account.balance=gt:x&account.balance=lte:y and ascending order param
- Added balances-04-publickey.spec.json to verify /api/v1/balances?account.publickey=x
- Added balances-05-timestamp.spec.json to verify /api/v1/balances?timestamp=<seconds>.<nanos>
- Removed duplicate valid_duration_seconds and valid_start_timestamp from transactions-01-no-args.spec.json
- Added transactions-04-timestamp.spec.json to verify /api/v1/transactions?timestamp=<seconds>.<nanos>
- Added transactions-05-success-result.spec.json to verify /api/v1/transactions?result=success
- Added transactions-06-fail-result.spec.json to verify /api/v1/transactions?result=fail
- Added transactions-07-credit-type.spec.json to verify /api/v1/transactions?account.id=x.y.z&type=credit
- Added transactions-08-debit-type.spec.json to verify /api/v1/transactions?account.id=x.y.z&type=debit

**Which issue(s) this PR fixes**:
Partially addresses #412

**Special notes for your reviewer**:
Will add negative and empty response cases in a new PR

**Checklist**
- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

**Comment by steven-sheehy** [comment]
Haven't reviewed yet, but few questions:

Does this cover all parameters?

Can we add negative test cases? Both non 2xx and empty response scenarios.

Can we investigate code coverage tools to see what our percentage is? Can be a follow up if it's non trivial.

**Comment by Nana-EC** [comment]
> Haven't reviewed yet, but few questions:
> 
> Does this cover all parameters?
> 
> Can we add negative test cases? Both non 2xx and empty response scenarios.
> 
> Can we investigate code coverage tools to see what our percentage is? Can be a follow up if it's non trivial.

Yes covers all parameters per out public API documentation.
Good suggestion will add more negative cases for HTTP responses and empty scenario.
Would like want to create a separate ticket for coverage tools, but I'll take a quick look.

File: hedera-mirror-rest/__tests__/specs/balances-03-balance-range.spec.json [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      {
+        "timestamp": 2345,
+        "id": 21,
+        "balance": 300
+      }
+    ],
+    "transactions": [],
+    "cryptotransfers": []
+  },
+  "url": "/api/v1/balances?account.balance=gt:300&account.balance=lte:700&&order=asc",
```
Comment (apeksharma) [inline hedera-mirror-rest/__tests__/specs/balances-03-balance-range.spec.json]: In followup, if you can come up with a simple way to have one common setup for accounts-02, accounts-03, accounts-04, balances-02, balances-03, balances-05, and for new tests - that'd be great. If not, it's okay.

Comment (Nana-EC) [reply hedera-mirror-rest/__tests__/specs/balances-03-balance-range.spec.json]: So it just happens to be that they are the same, it's not by design, more so me just not changing stuff. In fact it's probably better that they are all different so there's no chance of any sticky logic or a false positives.

**Comment by Nana-EC** [comment]
> This is already large enough. If code changes in this PR are not going to be affected, would be good if other tests can be added in separate PR.

Sounds good will do.

**Pull request merged**